common/build.gradle.kts

```diff
@@ -0,0 +1,41 @@
+import org.gradle.kotlin.dsl.named
+import org.jetbrains.kotlin.gradle.tasks.KotlinCompile
+
+repositories {
+    mavenCentral()
+}
+
+kotlin {
+    configureKotlin()
+}
+
+java {
+    configureJava()
+}
+
+sourceSets {
+    main {
+        java.srcDirs("src/main")
+    }
+
+    dependencies {
+        // main
+        val kotlinVersion: String by project
+
+        compileOnly(project(":bootstrap"))
+        api(project(":trace")) // TODO: can this be fixed?
```

Comment:
I think `common` should not depend on `trace`, it should be the other way around.